tests/lowering/eltwise/fused/test_batch_norm_inference.py

```diff
@@ -0,0 +1,61 @@
+import torch
+import torch_ttnn
+import pytest
+import ttnn
+
+
+class BatchNormModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, *args, **kwargs):
+        return torch.nn.functional.batch_norm(*args, **kwargs)
+
+
+def expand_param_tuple(*shape):
+    shape = (*shape,)
+    return (
+        (shape, True, True),
+        (shape, False, True),
+        (shape, True, False),
+        (shape, False, False),
+    )
+
+
+@pytest.mark.parametrize(
+    "input_shape, weight, bias",
+    (
+        *expand_param_tuple(1, 100, 14, 14),
+        *expand_param_tuple(1, 224, 112, 112),
+        *expand_param_tuple(1, 640, 14, 14),
+        *expand_param_tuple(1, 640, 7, 7),
+        *expand_param_tuple(1, 672, 15, 15),
+        *expand_param_tuple(1, 64, 30, 40),
+        *expand_param_tuple(2, 15, 30, 40),
```

Comment:
@jerrysky3, this particular case of batch = 2 fails, but I don't know why.

Reply:
```
E           RuntimeError: The size of tensor a (2) must match the size of tensor b (0) at non-singleton dimension 0

../tt-metal/python_env/lib/python3.8/site-packages/torch/_tensor.py:1418: RuntimeError
--------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------
arg0_1: {'val': FakeTensor(..., size=(2, 15, 30, 40), dtype=torch.bfloat16), 'tensor_meta': TensorMetadata(shape=torch.Size([2, 15, 30, 40]), dtype=torch.bfloat16, requires_grad=False, stride=(18000, 1200, 40, 1), memory_format=torch.contiguous_format, is_quantized=False, qparams={})}
arg1_1: {'val': FakeTensor(..., size=(15,), dtype=torch.bfloat16), 'tensor_meta': TensorMetadata(shape=torch.Size([15]), dtype=torch.bfloat16, requires_grad=False, stride=(1,), memory_format=torch.contiguous_format, is_quantized=False, qparams={})}
arg2_1: {'val': FakeTensor(..., size=(15,), dtype=torch.bfloat16), 'tensor_meta': TensorMetadata(shape=torch.Size([15]), dtype=torch.bfloat16, requires_grad=False, stride=(1,), memory_format=torch.contiguous_format, is_quantized=False, qparams={})}
1: 
2: 
3: 
4: def forward(self, arg0_1, arg1_1, arg2_1):
5:     ttnn_from_torch = ttnn_decorators_ttnn_from_torch(arg2_1, layout = ttnn_TILE_LAYOUT, device = ttnn_Specified_Device, dtype = ttnn_bfloat16);  arg2_1 = None
6:     ttnn_add = ttnn_decorators_ttnn_add(ttnn_from_torch, 1e-05);  ttnn_from_torch = None
7:     ttnn_rsqrt = ttnn_decorators_ttnn_rsqrt(ttnn_add);  ttnn_add = None
8:     ttnn_from_device = ttnn_decorators_ttnn_from_device(ttnn_rsqrt);  ttnn_rsqrt = None
9:     ttnn_to_layout = ttnn_decorators_ttnn_to_layout(ttnn_from_device, ttnn_ROW_MAJOR_LAYOUT);  ttnn_from_device = None
10:     ttnn_reshape = ttnn_decorators_ttnn_reshape(ttnn_to_layout, (1, 15, 1, 1));  ttnn_to_layout = None
11:     ttnn_from_torch_1 = ttnn_decorators_ttnn_from_torch(arg1_1, layout = ttnn_TILE_LAYOUT, device = ttnn_Specified_Device, dtype = ttnn_bfloat16);  arg1_1 = None
12:     ttnn_from_device_1 = ttnn_decorators_ttnn_from_device(ttnn_from_torch_1);  ttnn_from_torch_1 = None
13:     ttnn_to_layout_1 = ttnn_decorators_ttnn_to_layout(ttnn_from_device_1, ttnn_ROW_MAJOR_LAYOUT);  ttnn_from_device_1 = None
14:     ttnn_reshape_1 = ttnn_decorators_ttnn_reshape(ttnn_to_layout_1, (1, 15, 1, 1));  ttnn_to_layout_1 = None
15:     ttnn_from_torch_2 = ttnn_decorators_ttnn_from_torch(arg0_1, layout = ttnn_TILE_LAYOUT, device = ttnn_Specified_Device, dtype = ttnn_bfloat16);  arg0_1 = None
16:     ttnn_to_layout_2 = ttnn_decorators_ttnn_to_layout(ttnn_reshape_1, ttnn_TILE_LAYOUT);  ttnn_reshape_1 = None
17:     ttnn_to_device = ttnn_decorators_ttnn_to_device(ttnn_to_layout_2, device = ttnn_Specified_Device);  ttnn_to_layout_2 = None
18:     ttnn_subtract = ttnn_decorators_ttnn_subtract(ttnn_from_torch_2, ttnn_to_device);  ttnn_from_torch_2 = ttnn_to_device = None
19:     ttnn_to_layout_3 = ttnn_decorators_ttnn_to_layout(ttnn_reshape, ttnn_TILE_LAYOUT);  ttnn_reshape = None
20:     ttnn_to_device_1 = ttnn_decorators_ttnn_to_device(ttnn_to_layout_3, device = ttnn_Specified_Device);  ttnn_to_layout_3 = None
21:     ttnn_multiply = ttnn_decorators_ttnn_multiply(ttnn_subtract, ttnn_to_device_1);  ttnn_subtract = ttnn_to_device_1 = None
22:     getitem = ttnn_multiply[0];  ttnn_multiply = None
23:     ttnn_to_torch = ttnn_decorators_ttnn_to_torch(getitem);  getitem = None
24:     _to_copy_default = torch.ops.aten._to_copy.default(ttnn_to_torch, dtype = torch.bfloat16);  ttnn_to_torch = None
25:     return (_to_copy_default,)
26:     
opcode         name                target                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           args                                         kwargs
-------------  ------------------  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------  -------------------------------------------  -------------------------------------------------------------------------------------
placeholder    arg0_1              arg0_1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           ()                                           {}
placeholder    arg1_1              arg1_1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           ()                                           {}
placeholder    arg2_1              arg2_1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           ()                                           {}
call_function  ttnn_from_torch     FastOperation(python_fully_qualified_name='ttnn.from_torch', function=<function from_torch at 0x7f3f58dc4ee0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4dc0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                                             (arg2_1,)                                    {'layout': ttnn_TILE_LAYOUT, 'device': ttnn_Specified_Device, 'dtype': ttnn_bfloat16}
call_function  ttnn_add            FastOperation(python_fully_qualified_name='ttnn.add', function=<ttnn._ttnn.operations.binary.add_t object at 0x7f3fc3c08b30>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc2a60>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                               (ttnn_from_torch, 1e-05)                     {}
call_function  ttnn_rsqrt          FastOperation(python_fully_qualified_name='ttnn.rsqrt', function=<ttnn._ttnn.operations.unary.rsqrt_t object at 0x7f3fc3c2d530>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function register_ttnn_cpp_unary_function.<locals>._golden_function at 0x7f3f58d85670>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)  (ttnn_add,)                                  {}
call_function  ttnn_from_device    FastOperation(python_fully_qualified_name='ttnn.from_device', function=<built-in method from_device of PyCapsule object at 0x7f3fc3c13510>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58d51160>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                (ttnn_rsqrt,)                                {}
call_function  ttnn_to_layout      FastOperation(python_fully_qualified_name='ttnn.to_layout', function=<ttnn._ttnn.operations.core.to_layout_t object at 0x7f3fc3c14370>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58d51280>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                     (ttnn_from_device, ttnn_ROW_MAJOR_LAYOUT)    {}
call_function  ttnn_reshape        FastOperation(python_fully_qualified_name='ttnn.reshape', function=<ttnn._ttnn.operations.data_movement.reshape_t object at 0x7f3fc3bad8b0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4ca0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                (ttnn_to_layout, (1, 15, 1, 1))              {}
call_function  ttnn_from_torch_1   FastOperation(python_fully_qualified_name='ttnn.from_torch', function=<function from_torch at 0x7f3f58dc4ee0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4dc0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                                             (arg1_1,)                                    {'layout': ttnn_TILE_LAYOUT, 'device': ttnn_Specified_Device, 'dtype': ttnn_bfloat16}
call_function  ttnn_from_device_1  FastOperation(python_fully_qualified_name='ttnn.from_device', function=<built-in method from_device of PyCapsule object at 0x7f3fc3c13510>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58d51160>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                (ttnn_from_torch_1,)                         {}
call_function  ttnn_to_layout_1    FastOperation(python_fully_qualified_name='ttnn.to_layout', function=<ttnn._ttnn.operations.core.to_layout_t object at 0x7f3fc3c14370>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58d51280>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                     (ttnn_from_device_1, ttnn_ROW_MAJOR_LAYOUT)  {}
call_function  ttnn_reshape_1      FastOperation(python_fully_qualified_name='ttnn.reshape', function=<ttnn._ttnn.operations.data_movement.reshape_t object at 0x7f3fc3bad8b0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4ca0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                (ttnn_to_layout_1, (1, 15, 1, 1))            {}
call_function  ttnn_from_torch_2   FastOperation(python_fully_qualified_name='ttnn.from_torch', function=<function from_torch at 0x7f3f58dc4ee0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4dc0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                                             (arg0_1,)                                    {'layout': ttnn_TILE_LAYOUT, 'device': ttnn_Specified_Device, 'dtype': ttnn_bfloat16}
call_function  ttnn_to_layout_2    FastOperation(python_fully_qualified_name='ttnn.to_layout', function=<ttnn._ttnn.operations.core.to_layout_t object at 0x7f3fc3c14370>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58d51280>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                     (ttnn_reshape_1, ttnn_TILE_LAYOUT)           {}
call_function  ttnn_to_device      FastOperation(python_fully_qualified_name='ttnn.to_device', function=<built-in method to_device of PyCapsule object at 0x7f3fc3c134e0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4f70>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                    (ttnn_to_layout_2,)                          {'device': ttnn_Specified_Device}
call_function  ttnn_subtract       FastOperation(python_fully_qualified_name='ttnn.subtract', function=<ttnn._ttnn.operations.binary.subtract_t object at 0x7f3fc3bd84b0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc2af0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                     (ttnn_from_torch_2, ttnn_to_device)          {}
call_function  ttnn_to_layout_3    FastOperation(python_fully_qualified_name='ttnn.to_layout', function=<ttnn._ttnn.operations.core.to_layout_t object at 0x7f3fc3c14370>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58d51280>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                     (ttnn_reshape, ttnn_TILE_LAYOUT)             {}
call_function  ttnn_to_device_1    FastOperation(python_fully_qualified_name='ttnn.to_device', function=<built-in method to_device of PyCapsule object at 0x7f3fc3c134e0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4f70>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                    (ttnn_to_layout_3,)                          {'device': ttnn_Specified_Device}
call_function  ttnn_multiply       FastOperation(python_fully_qualified_name='ttnn.multiply', function=<ttnn._ttnn.operations.binary.multiply_t object at 0x7f3fc3bd8bf0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc2b80>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=True, is_experimental=False)                                     (ttnn_subtract, ttnn_to_device_1)            {}
call_function  getitem             <built-in function getitem>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      (ttnn_multiply, 0)                           {}
call_function  ttnn_to_torch       FastOperation(python_fully_qualified_name='ttnn.to_torch', function=<function to_torch at 0x7f3f58d510d0>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7f3fc36e38b0>, golden_function=<function _golden_function at 0x7f3f58dc4e50>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7f3fc36e3af0>, is_cpp_operation=False, is_experimental=False)                                                                 (getitem,)                                   {}
call_function  _to_copy_default    aten._to_copy.default                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            (ttnn_to_torch,)                             {'dtype': torch.bfloat16}
output         output              output                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           ((_to_copy_default,),)                       {}
------------------------------------------------------------------------- Captured stdout teardown --------------------------------------------------------------------------
                  Metal | INFO     | Closing device 2
                  Metal | INFO     | Disabling and clearing program cache on device 2
========================================================================== short test summary info ==========================================================================
FAILED tests/lowering/eltwise/fused/test_batch_norm_inference.py::test_batch_norm_inference[input_shape24-True-True] - RuntimeError: The size of tensor a (2) must match the size of tensor b (0) at non-singleton dimension 0
FAILED tests/lowering/eltwise/fused/test_batch_norm_inference.py::test_batch_norm_inference[input_shape25-False-True] - RuntimeError: The size of tensor a (2) must match the size of tensor b (0) at non-singleton dimension 0
FAILED tests/lowering/eltwise/fused/test_batch_norm_inference.py::test_batch_norm_inference[input_shape26-True-False] - RuntimeError: The size of tensor a (2) must match the size of tensor b (0) at non-singleton dimension 0
FAILED tests/lowering/eltwise/fused/test_batch_norm_inference.py::test_batch_norm_inference[input_shape27-False-False] - RuntimeError: The size of tensor a (2) must match the size of tensor b (0) at non-singleton dimension 0
======================================================================= 4 failed, 24 passed in 15.82s =======================================================================
                 Device | INFO     | Closing user mode device drivers
```

Reply:
This case seems to be passed now (tested on rc47), might be fixed in the upstream?